src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystemContract;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemAdapter;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+class TracingFilesystem implements CloudFilesystemContract
+{
+    protected $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @template TResult
+     * @param array<string, mixed> $data
+     * @param callable(): TResult $originalCall
+     * @return TResult
+     */
+    protected function withTracing(string $method, array $data, callable $originalCall)
+    {
+        $context = new SpanContext();
+        $context->setOp("storage.{$method}");
+        $context->setData($data);
+
+        return trace($originalCall, $context);
+    }
+
+    public function assertExists($path, $content = null)
+    {
+        return $this->withTracing('assertExists', ['path' => $path], function () use ($path, $content) {
+            // TODO assertExists only present on FilesystemAdapter
+            return $this->filesystem->assertExists($path, $content);
```

Comment:
Should we require `FilesystemAdapter` to be used so this always works? Alternatively, we could throw if `! $this->filesystem instanceof FilesystemAdapter` in those particular calls.

Reply:
I made this throw.

Reply:
In general, we try to avoid the SDK breaking customer applications. While I agree that some issues might be hard to debug, especially in PHP, we must always degrade gracefully. In this case, just no-op.

Reply:
In this case, the application would fail without the SDK being present too.

Reply:
It's fine if an exception is thrown by the framework, we should just refrain from throwing one ourselves.

Reply:
I see, so we just try the call without checking? I guess it could still work due to duck typing, so we can let it slide.